### ISSUE TITLE
508-defect-2: Groups of form elements must have group labels (e.g. <fieldset> and <legend>)

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/va.gov-team#55444]
### Point of contact

@jeana-adhoc

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

Groups of form elements MUST have group labels (e.g. `<fieldset>` and `<legend>`, to meet Section 508 requirements. 

The service connected conditions question (last question) of the form has the <legend> wrapped around the incorrect information.

### Reproduction steps

1. Enter https://staging.va.gov/housing-assistance/disability-housing-grants/adapted-housing/ in a browser
2. Continue through the form until you get to "Service-connected conditions"
3. Start a screen reader and tab to the question
4. Verify that there isn't a semantic correlation between the checkboxes and the question "Do you have any of these conditions that were caused—or made worse—by your service?"

### Proposed solution or next steps

The `<h3>Service-connected conditions</h3>` should be moved out of the fieldset. Related to this, the `vads-u-font-size--h2` class should be removed from the `<h3>` as well.

Then, `<p>Do you have any of these conditions that were caused—or made worse—by your service?</p>` Should be wrapped in the `<legend>` tag.

### References, articles, or WCAG support

[WCAG 2.0 Level A - 1.3.1 - Info and Relationships: Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text.](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

[Wuhcag 1.3.1 – Info and Relationships (Level A)](https://www.wuhcag.com/info-and-relationships/)

[WCAG 2.0 Level A - 4.1.1 - Parsing](http://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-parses.html)

### Type of issue

- [X] Screenreader
- [X] Keyboard
- [ ] Focus
- [X] Headings
- [ ] Color
- [ ] Zoom
- [X] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by tlei123** [comment]
@jeana-adhoc @ndsprinkle , I don't think Forms Library lets us implement the "Proposed solution" above.  I couldn't find any way to either (a) turn off wrapping our `ui:title` [`<h3>`] in a `<legend>`, or (b) specify a different element to use as `<legend>`.

However, what I can do is add a copy of the question's text as a "screen-reader-only" `<span>` to our existing `<h3`, so that screen-readers would read it right after "Service-connected conditions":
```html
<h3 className="vads-u-color--gray-dark vads-u-margin-y--0">
  Service-connected conditions
  <span className="sr-only">
    : Do you have any of these conditions that were caused&mdash;or made
    worse&mdash;by your service?
  </span>
</h3>
```

Here's a link to a video of my local MacOS VoiceOver:
https://user-images.githubusercontent.com/587583/229661587-1ed49bb2-ed7c-44ba-a49b-f030d0d947dc.mp4

Let me know what you think.

**Comment by jeana-adhoc** [comment]
Hi @tlei123 - This is a great work around. And would technically get the job done. But, I wonder if we could structure the question like we do in `/26-4555/pages/previousSahApplication1.js`

It has a similar structure. An `H3` followed by a radio button group. (I know this question has checkboxes instead, so maybe checkboxes and radio buttons can't be treated the same). But, is this a possibility?

**Comment by tlei123** [comment]
I think I've got what you wanted.  Let me know if this works for you...

Here's a link to a video of my local MacOS VoiceOver:

https://user-images.githubusercontent.com/587583/229906614-bf80cd73-4cce-48d9-b50b-e2df4697fe28.mp4


Thanks to Nick's guidance, I've re-structured the form so that the old `ui:description` now becomes a separate `ui:title` for the `<fieldset>`:

```js
uiSchema: {
  'ui:title': ( // this title moved out of remarks object, so won't get wrapped in <legend>
    <h3 className="vads-u-color--gray-dark vads-u-margin-y--0">
      Service-connected conditions
    </h3>
  ),
  remarks: {
    'ui:title': ( // this title was description before, now as title gets wrapped in <legend>
      <p className="vads-u-font-family--sans vads-u-font-size--base vads-u-font-weight--normal">
        Do you have any of these conditions that were caused&mdash;or made
        worse&mdash;by your service?
      </p>
    ),
    respiratoryIllness: {
      'ui:title': 'A severe respiratory (breathing-related) illness',
    },
    ...
  }
}
```
 
cc @ndsprinkle [using a separate `ui:title` for remarks object worked]

**Comment by jeana-adhoc** [comment]
@tlei123 - This fix looks good to me. I also confirmed it in my own environment.

**Comment by tlei123** [comment]
@ndsprinkle , [PR opened](https://github.com/department-of-veterans-affairs/vets-website/pull/23842) for your review & approval.  :)

**Comment by tlei123** [comment]
PR merged; will validate in Staging tmrw.

**Comment by tlei123** [comment]
Validated successfully on Staging.
